Pages/Loginpage.cs

```diff
@@ -1,41 +1,51 @@
-﻿using OpenQA.Selenium;
+﻿using Marsproject.Support;
+using OpenQA.Selenium;
 using OpenQA.Selenium.Chrome;
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
 using System.Text;
 using System.Threading.Tasks;
 
 namespace Marsproject.Pages
 {
-    public class Loginpage
-    {
+    public class Loginpage : NewCommondriver
 
-        public By SignInButton = By.XPath("//*[contains(text(),'Sign')]");
+    {
+        public Loginpage(IWebDriver driver)
+        {
+            this.driver = driver;
+        }
+        //  public By SignInButton = By.XPath("//*[contains(text(),'Sign')]");
 
-        public By Email_field = By.Name("email");
+        //public By Email_field = By.Name("email");
 
-        public By Password_field = By.Name("password");
+        //public By Password_field = By.Name("password");
 
-        public By Loginbutton = By.XPath("//*[contains(text(),'Login')]");
+        //public By Loginbutton = By.XPath("//*[contains(text(),'Login')]");
 
         public By homepage_text = By.XPath("//*[contains(text(),'Mars Logo')]");
 
         public By EmailID_field = By.Name("email");
 
-       
-        public void Loginwith_INvalid_cerdentials(IWebDriver driver, string Email_ID, string Password)
+
+        public void Loginwith_INvalid_cerdentials(string Email_ID, string Password)
         {
-           
+
```

Comment:
driver = new ChromeDriver(); this  should not be here in your NewCommondriver class you should initialise the driver and then also expose it so that when you extend to it the driver is available here! the navigate to happen there are well! I suggest you create a Init method which does both and in the Hooks class create a variable, where then you will use in the before hooks to call the init method!

Reply:
please remove all the place where you are doing   driver = new ChromeDriver();

            driver.Navigate().GoToUrl("http://localhost:5000/");
            driver.Manage().Window.Maximize();